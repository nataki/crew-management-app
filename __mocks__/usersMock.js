export const users = [
    {
        "name": {
            "title": "ms",
            "first": "maya",
            "last": "davies"
        },
        "location": {
            "city": "chichester"
        },
        "id": {
            "name": "NINO",
            "value": "KZ 97 53 12 J"
        },
        "picture": {
            "thumbnail": "https://randomuser.me/api/portraits/thumb/women/37.jpg"
        },
        "status": "USER_STATUS_INTERVIEWED"
    },
    {
        "name": {
            "title": "mr",
            "first": "carlos",
            "last": "lynch"
        },
        "location": {
            "city": "plymouth"
        },
        "id": {
            "name": "NINO",
            "value": "LH 84 58 54 N"
        },
        "picture": {
            "thumbnail": "https://randomuser.me/api/portraits/thumb/men/27.jpg"
        },
        "status": "USER_STATUS_APPLIED"
    },
    {
        "name": {
            "title": "mrs",
            "first": "isabella",
            "last": "perez"
        },
        "location": {

            "city": "truro"
        },
        "id": {
            "name": "NINO",
            "value": "LB 87 76 59 X"
        },
        "picture": {

            "thumbnail": "https://randomuser.me/api/portraits/thumb/women/67.jpg"
        },
        "status": "USER_STATUS_APPLIED"
    },
    {
        "name": {
            "title": "mrs",
            "first": "vicki",
            "last": "nelson"
        },
        "location": {
            "city": "truro"
        },
        "id": {
            "name": "NINO",
            "value": "JR 06 90 73 J"
        },
        "picture": {
            "thumbnail": "https://randomuser.me/api/portraits/thumb/women/27.jpg"
        },
        "status": "USER_STATUS_HIRED"
    },
    {
        "gender": "female",
        "name": {
            "title": "mrs",
            "first": "ellie",
            "last": "wilson"
        },
        "location": {
            "street": "9737 park lane",
            "city": "armagh"
        },
        "id": {
            "name": "NINO",
            "value": "NS 40 06 52 K"
        },
        "picture": {
            "thumbnail": "https://randomuser.me/api/portraits/thumb/women/32.jpg"
        },
        "status": "USER_STATUS_HIRED"
    }
];