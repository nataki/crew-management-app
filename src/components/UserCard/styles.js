export default {
    user_card: {
        border: '1px solid grey',
        boxShadow: '0 4px 8px 0 rgba(0,0,0,0.2)',
        borderRadius: '3px',
        marginBottom: '10px',
    },
    user_info: {
        display: 'flex',
    },
    user_photo: {
        marginRight: '10px',
        padding: '5px'
    },
    user_name: {
        '& div': {
            fontSize: '0.8em',
            fontStyle: 'italic',
            marginTop: '10px'
        }
    },
    controls: {
        display: 'flex',
        justifyContent: 'space-between',
        padding: '5px'
    }
}