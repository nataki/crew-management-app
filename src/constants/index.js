export const STATUS_LOADING = 'STATUS_LOADING';
export const STATUS_LOADED = 'STATUS_LOADED';
export const USER_STATUS_APPLIED = 'USER_STATUS_APPLIED';
export const USER_STATUS_INTERVIEWED = 'USER_STATUS_INTERVIEWED';
export const USER_STATUS_HIRED = 'USER_STATUS_HIRED';