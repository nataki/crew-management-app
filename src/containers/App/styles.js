export default {
    app: {
    },
    header: {
        display: 'flex'
    },
    container: {
        display: 'flex'
    }
}