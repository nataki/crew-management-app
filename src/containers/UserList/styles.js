export default {
    column: {
        flex: '1',
        padding: '20px',
    },
    category_title: {
        fontWeight: 'bold',
        lineHeight: '40px',
        textAlign: 'center'
    }
}