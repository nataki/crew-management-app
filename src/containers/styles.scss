.app {
    .header{
        display: flex;
    }
    .container{
        display: flex;
        .column {
            flex: 1;
            padding: 20px;

            .category_title{
                text-align: center;
                font-weight: bold;
                line-height: 40px;
            }
        }
    }
}

.user_card {
    border: 1px solid grey;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    border-radius: 3px;
    margin-bottom: 10px;
    .user_info {
        display: flex;
        .user_photo {
            margin-right: 10px;
            padding: 5px;
        }
        .user_name div {
            font-size: 0.8em;
            font-style: italic;
            margin-top:10px;
        }
        .controls {
            display: flex;
            justify-content: space-between;
            padding: 5px;
        }
    }
}