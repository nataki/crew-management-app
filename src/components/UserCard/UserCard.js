// @flow
import React from 'react';
import injectSheet from 'react-jss';
import {compose} from 'recompose';
import styles from './styles';

type Props = {
    classes: { user_card: string, user_info: string, user_photo: string, user_name: string, controls: string},
    id: string,
    image: string,
    name: string,
    location: string,
    onLeftClick: () => void,
    onRightClick: () => void,
}

const UserCard = ({classes, id, image, name, location, onLeftClick, onRightClick}: Props) => (
    <div className={classes.user_card}>
        <div className={classes.user_info}>
            <div className={classes.user_photo}>
                <img src={image} alt={name} />
            </div>
            <div className={classes.user_name}>
                {name}
                <div>{location}</div>
            </div>
        </div>
        <div className={classes.controls}>
            {onLeftClick && <button onClick={onLeftClick}>Left</button>}
            {onRightClick && <button onClick={onRightClick}>Right</button>}
        </div>
    </div>
);

export default compose(
    injectSheet(styles)
)(UserCard);