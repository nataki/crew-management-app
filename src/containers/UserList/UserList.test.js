import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import ConnectedUserList, {UserList} from './UserList';
import {STATUS_LOADED} from "../../constants/index";
import {users} from "../../../__mocks__/usersMock";
import UserCard from "../../components/UserCard/index";
import styles from "./styles";

const classes = Object.keys(styles)
    .reduce(
        (prev, curr) => ({
            ...prev,
            [curr]: curr
        }),
        {}
    );

describe('UserList - snapshot',()=>{
    it('capturing snapshot of UserList', () => {
        const shallowRenderer = new ShallowRenderer();
        shallowRenderer.render(<UserList classes={classes} />);
        const renderedValue = shallowRenderer.getRenderOutput();
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('Connected UserList - snapshot',() => {
    const initialState = {
        status: STATUS_LOADED,
        filters: {city: "", name: ""},
        users: users
    };
    const mockStore = configureStore();
    const shallowRenderer = new ShallowRenderer();
    let store, container;

    beforeEach(() => {
        store = mockStore(initialState);
    });

    it('pass properties from state', () => {
        shallowRenderer.render(<ConnectedUserList classes={classes} store={store} />);
        container = shallowRenderer.getRenderOutput();

        expect(container.props.appliedUsers.length).toEqual(2);
        expect(container.props.interviewedUsers.length).toEqual(1);
        expect(container.props.hiredUsers.length).toEqual(2);
    });

    it('match snapshot', () => {
        const testRenderer = renderer.create(<ConnectedUserList classes={classes} store={store} />).toJSON();
        expect(testRenderer).toMatchSnapshot();
    });


    it('render usercards by status', () => {
        const testRenderer = renderer.create(<ConnectedUserList classes={classes} store={store} />);
        const testInstance = testRenderer.root;
        const columns = testInstance.findAllByProps({className: "column"});

        expect(columns.length).toEqual(3);
        expect(columns[0].findAllByType(UserCard).length).toEqual(2);
        expect(columns[1].findAllByType(UserCard).length).toEqual(1);
        expect(columns[2].findAllByType(UserCard).length).toEqual(2);
    });
});
