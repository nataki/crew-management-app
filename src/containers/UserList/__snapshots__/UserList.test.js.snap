// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected UserList - snapshot match snapshot 1`] = `
Array [
  <div
    className="column"
  >
    <div
      className="category_title"
    >
      Applied
    </div>
    <div
      className="UserCard-user_card-0-1-3"
    >
      <div
        className="UserCard-user_info-0-1-4"
      >
        <div
          className="UserCard-user_photo-0-1-5"
        >
          <img
            alt="mr carlos lynch"
            src="https://randomuser.me/api/portraits/thumb/men/27.jpg"
          />
        </div>
        <div
          className="UserCard-user_name-0-1-6"
        >
          mr carlos lynch
          <div>
            plymouth
          </div>
        </div>
      </div>
      <div
        className="UserCard-controls-0-1-7"
      >
        <button
          onClick={[Function]}
        >
          Right
        </button>
      </div>
    </div>
    <div
      className="UserCard-user_card-0-1-3"
    >
      <div
        className="UserCard-user_info-0-1-4"
      >
        <div
          className="UserCard-user_photo-0-1-5"
        >
          <img
            alt="mrs isabella perez"
            src="https://randomuser.me/api/portraits/thumb/women/67.jpg"
          />
        </div>
        <div
          className="UserCard-user_name-0-1-6"
        >
          mrs isabella perez
          <div>
            truro
          </div>
        </div>
      </div>
      <div
        className="UserCard-controls-0-1-7"
      >
        <button
          onClick={[Function]}
        >
          Right
        </button>
      </div>
    </div>
  </div>,
  <div
    className="column"
  >
    <div
      className="category_title"
    >
      Interviewed
    </div>
    <div
      className="UserCard-user_card-0-1-3"
    >
      <div
        className="UserCard-user_info-0-1-4"
      >
        <div
          className="UserCard-user_photo-0-1-5"
        >
          <img
            alt="ms maya davies"
            src="https://randomuser.me/api/portraits/thumb/women/37.jpg"
          />
        </div>
        <div
          className="UserCard-user_name-0-1-6"
        >
          ms maya davies
          <div>
            chichester
          </div>
        </div>
      </div>
      <div
        className="UserCard-controls-0-1-7"
      >
        <button
          onClick={[Function]}
        >
          Left
        </button>
        <button
          onClick={[Function]}
        >
          Right
        </button>
      </div>
    </div>
  </div>,
  <div
    className="column"
  >
    <div
      className="category_title"
    >
      Hired
    </div>
    <div
      className="UserCard-user_card-0-1-3"
    >
      <div
        className="UserCard-user_info-0-1-4"
      >
        <div
          className="UserCard-user_photo-0-1-5"
        >
          <img
            alt="mrs vicki nelson"
            src="https://randomuser.me/api/portraits/thumb/women/27.jpg"
          />
        </div>
        <div
          className="UserCard-user_name-0-1-6"
        >
          mrs vicki nelson
          <div>
            truro
          </div>
        </div>
      </div>
      <div
        className="UserCard-controls-0-1-7"
      >
        <button
          onClick={[Function]}
        >
          Left
        </button>
      </div>
    </div>
    <div
      className="UserCard-user_card-0-1-3"
    >
      <div
        className="UserCard-user_info-0-1-4"
      >
        <div
          className="UserCard-user_photo-0-1-5"
        >
          <img
            alt="mrs ellie wilson"
            src="https://randomuser.me/api/portraits/thumb/women/32.jpg"
          />
        </div>
        <div
          className="UserCard-user_name-0-1-6"
        >
          mrs ellie wilson
          <div>
            armagh
          </div>
        </div>
      </div>
      <div
        className="UserCard-controls-0-1-7"
      >
        <button
          onClick={[Function]}
        >
          Left
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`UserList - snapshot capturing snapshot of UserList 1`] = `
<React.Fragment>
  <div
    className="column"
  >
    <div
      className="category_title"
    >
      Applied
    </div>
  </div>
  <div
    className="column"
  >
    <div
      className="category_title"
    >
      Interviewed
    </div>
  </div>
  <div
    className="column"
  >
    <div
      className="category_title"
    >
      Hired
    </div>
  </div>
</React.Fragment>
`;
